src/VisualStudio/Core/Def/Implementation/Preview/FileChange.cs

```diff
@@ -225,14 +231,51 @@ internal override void GetDisplayData(VSTREEDISPLAYDATA[] pData)
             pData[0].Image = pData[0].SelectedImage = (ushort)StandardGlyphGroup.GlyphLibrary;
         }
 
-        public static IHierarchicalDifferenceCollection ComputeDiffSpans(SourceText oldText, SourceText newText, ITextDifferencingService diffService, CancellationToken cancellationToken)
+        private static IHierarchicalDifferenceCollection ComputeDiffSpans(ITextDifferencingService diffService, TextDocument left, TextDocument right, CancellationToken cancellationToken)
         {
-            var diffResult = diffService.DiffStrings(oldText.ToString(), newText.ToString(), new StringDifferenceOptions()
+            // TODO: it would be nice to have a syntax based differ for presentation here, 
+            //       current way of just using text differ has its own issue, and using syntax differ in compiler that are for incremental parser
+            //       has its own drawbacks.
+
+            var oldText = left.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var newText = right.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+
+            var oldString = oldText.ToString();
+            var newString = newText.ToString();
+
+            // first try, cheapest way.
+            var diffResult = diffService.DiffStrings(oldString, newString, new StringDifferenceOptions()
             {
-                DifferenceType = StringDifferenceTypes.Line
+                DifferenceType = StringDifferenceTypes.Line | StringDifferenceTypes.Word,
+                WordSplitBehavior = WordSplitBehavior.WhiteSpaceAndPunctuation
             });
 
-            return diffResult;
+            if (!ContainsBetterDiff(left, right, diffResult, cancellationToken))
+            {
+                return diffResult;
+            }
+
+            // second, try a bit more expansive way
+            return diffService.DiffStrings(oldString, newString, new StringDifferenceOptions()
+            {
+                DifferenceType = StringDifferenceTypes.Word,
+                WordSplitBehavior = WordSplitBehavior.WhiteSpaceAndPunctuation
+            });
+        }
+
+        private static bool ContainsBetterDiff(TextDocument left, TextDocument right, IHierarchicalDifferenceCollection diffResult, CancellationToken cancellationToken)
+        {
+            var textDiffCount = diffResult.Differences.Count;
+
+            var leftDocument = left as Document;
+            var rightDocument = right as Document;
+            if (leftDocument == null || rightDocument == null)
+            {
+                return false;
+            }
+
+            var syntaxDiffCount = rightDocument.GetTextChangesAsync(leftDocument, cancellationToken).WaitAndGetResult(cancellationToken).Count();
+            return syntaxDiffCount > textDiffCount;
```

Comment:
@heejaechang I think the check boxes should always be at granularity of line. Anything less than that would feel noisy imho. Line granularity would probably also match Dev12?

Once we have syntax based differ, we could go down to statement / expression level. But until then we should probably leave it at line level...